### PR TITLE
<form should default to POST method.

### PR DESCRIPTION
**Pull request by adashst**
This PR implements https://allthings.atlassian.net/browse/APP-2798.
Form method defaults to POST.

Checklist: 
- [x] Test added / Snapshots updated
- [x] Documentation added / updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "prepublishOnly": "echo 'To publish this, go to build/ and run 'yarn publish''",
     "watch:test": "yarn test --watch",
     "test": "NODE_ENV=test jest src/ scripts/",
+    "test:single": "NODE_ENV=test jest",
```
Comment (dan-lee) [inline package.json]: What is this used for?

File: src/molecules/Form/Form.jsx [hunk 2]
```diff
@@ -109,11 +111,13 @@ Also see the <a href="/molecules/TextInput/">TextInput</a> for allowed props.
 class Form extends React.Component {
   static propTypes = {
     children: PropTypes.node,
+    method: PropTypes.string,
     onSubmit: PropTypes.func.isRequired,
   }
 
   static defaultProps = {
     onSubmit: _ => _,
+    method: 'POST',
```
Comment (dan-lee) [inline src/molecules/Form/Form.jsx]: Couldn't this have any side effects on forms behaviour when submitting? I mean that it actually sends you to another page without `preventDefault` on form submission.

Check this CodeSandbox: https://codesandbox.io/s/92z69162qy

Comment (dan-lee) [reply src/molecules/Form/Form.jsx]: Oh never mind, our `handleSubmit` actually calls `preventDefault`. 👍

**Comment by dan-lee** [comment]
As we call `preventDefault` in our submit handler it should never actually send the form data as GET parameters. Do we really need this? 🤔

File: package.json (hunk 1 above)
Comment (adashst) [reply package.json]: I added this to test a single file.
If i do`yarn test Form.test` all tests are running.
With `yarn test:single Form.test`i can test just the `Form.test.js`

**Comment by adashst** [comment]
@dan-lee https://allthings.atlassian.net/browse/APP-2546

**Comment by dan-lee** [comment]
I saw that but couldn't figure out why it's a problem even with `preventDefault`.
But the discussion earlier cleared it up for me:
When the page is correctly loaded via SSR but the client code throws an error, the `preventDefault` is not triggered and default method `GET` for the form submission is used 💡

File: package.json (hunk 1 above)
Comment (dan-lee) [reply package.json]:
```suggestion
    "jest": "NODE_ENV=test jest",
```

What about that? Executes the following for me:

>$ yarn jest src/molecules/FloatingButton.test.js 
>$ NODE_ENV=test jest src/molecules/FloatingButton.test.js

Comment (dan-lee) [reply package.json]: Or I think better would be:

```
"test": "NODE_ENV=test jest",
"test:all": "yarn test src/ scripts/"
```

Then you could still use `yarn test src/MyFile.jsx`

Comment (adashst) [reply package.json]: Think i like it.
So `yarn test` still behaves the same and `yarn jest Form.test` would just run the given test (if provided). Right?

File: package.json [hunk 3]
```diff
@@ -37,7 +37,8 @@
     "test:example": "jest ./scripts/example.test.js",
     "prepublishOnly": "echo 'To publish this, go to build/ and run 'yarn publish''",
     "watch:test": "yarn test --watch",
-    "test": "NODE_ENV=test jest src/ scripts/",
+    "test": "yarn jest src/ scripts/",
+    "jest": "NODE_ENV=test jest",
```
Comment (adashst) [inline package.json]: @dan-lee Think this 👆🏻is ok?

Comment (dan-lee) [reply package.json]: Great!

**Pull request merged**